### PR TITLE
 [EOC-47] showing list content 

### PR DESCRIPTION
**Pull request by adamklepacz**

### CODE REVIEW

File: client/src/common/utils/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Convert any array that contains
+ * object with _id property
+ * to map with key as _id
+ * eg. const arr = [ {_id: 1213, name: 'example'} ]  ==>  { 1213: {_id: 1213, name: 'example'} }
+ *  */
+
+export const convertArrayToMap = array =>
```
Comment (marekrozmus) [inline client/src/common/utils/index.js]: https://lodash.com/docs/4.17.11#keyBy ?

File: client/src/modules/shopping-list/model/actions.js [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
 
 // Dispatchers
-export const fetchProducts = () => dispatch =>
-  getData(`${ENDPOINT_URL}/items`)
+export const fetchItemsFromGivenList = listId => dispatch =>
+  getData(`${ENDPOINT_URL}/shopping-lists/${listId}/get-products`)
     .then(resp => resp.json())
-    .then(json => dispatch(recieveProducts(json)))
+    .then(json => {
+      const { products } = json[0];
```
Comment (marekrozmus) [inline client/src/modules/shopping-list/model/actions.js]: ???

File: client/src/modules/shopping-list/model/actions.js [hunk 3]
```diff
@@ -1,30 +1,34 @@
...
+const fetchShoppingListMetaDataSuccess = data => ({
+  type: ShoppingListActionTypes.FETCH_META_DATA_SUCCESS,
   payload: data
 });
 
 // Dispatchers
-export const fetchProducts = () => dispatch =>
-  getData(`${ENDPOINT_URL}/items`)
+export const fetchItemsFromGivenList = listId => dispatch =>
+  getData(`${ENDPOINT_URL}/shopping-lists/${listId}/get-products`)
```
Comment (marekrozmus) [inline client/src/modules/shopping-list/model/actions.js]: remove 'get-'

File: server/controllers/shoppingList.js [hunk 4]
```diff
@@ -49,8 +50,113 @@ const getShoppingListById = (req, resp) => {
...
+    }
+  );
+};
+
+const updateShoppingListItem = (req, resp) => {
+  const { isOrdered, itemId, voterIds } = req.body;
+  const { id: listId } = req.params;
+
+  /**
+   * Create object with properties to update, but onyl these which
```
Comment (marekrozmus) [inline server/controllers/shoppingList.js]: 'onyl'

File: client/src/modules/shopping-list/components/InputBar/InputBar.jsx [hunk 5]
```diff
@@ -69,17 +76,20 @@ class InputBar extends Component {
 
 InputBar.propTypes = {
   currentUser: UserPropType.isRequired,
+  match: PropTypes.objectOf(PropTypes.any),
```
Comment (alemikolo) [inline client/src/modules/shopping-list/components/InputBar/InputBar.jsx]: You can declare match prop with details and specify all needed params types:  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
...
    })
  }).isRequired,

File: client/src/modules/shopping-list/components/ProductsList/ProductsList.jsx [hunk 6]
```diff
@@ -88,6 +99,7 @@ class ProductsList extends Component {
 
 ProductsList.propTypes = {
   currentUser: UserPropType.isRequired,
+  match: PropTypes.objectOf(PropTypes.any),
```
Comment (alemikolo) [inline client/src/modules/shopping-list/components/ProductsList/ProductsList.jsx]: mach declaration: see above

File: client/src/modules/shopping-list/index.jsx [hunk 7]
```diff
@@ -2,34 +2,40 @@ import React, { Component, Fragment } from 'react';
...
   };
 
   render() {
-    const { fetchStatus, products } = this.props;
-
-    const archiveList = products.filter(product => product.isOrdered);
-    const shoppingList = products.filter(product => !product.isOrdered);
+    const { fetchStatus, list } = this.props;
+    const listItems = list && list.products ? list.products : [];
+    const archiveList = listItems.filter(product => product.isOrdered);
```
Comment (alemikolo) [inline client/src/modules/shopping-list/index.jsx]: maybe change product to item? (line 38. - the same)

File: client/src/modules/shopping-list/index.jsx [hunk 8]
```diff
@@ -58,17 +64,20 @@ class ShoppingList extends Component {
 
 ShoppingList.propTypes = {
   fetchStatus: StatusPropType.isRequired,
-  products: PropTypes.arrayOf(PropTypes.object),
+  list: PropTypes.objectOf(PropTypes.any),
+  match: PropTypes.objectOf(PropTypes.any),
```
Comment (alemikolo) [inline client/src/modules/shopping-list/index.jsx]: match declaration: see above

File: client/src/modules/shopping-list/model/actions.js [hunk 9]
```diff
@@ -1,30 +1,34 @@
...
-export const recieveProducts = json => ({
-  type: ShoppingListActionTypes.FETCH_PRODUCTS_REQUEST,
-  products: json
+export const fetchProductsSuccess = (json, listId) => ({
+  type: ShoppingListActionTypes.FETCH_PRODUCTS_SUCCESS,
+  payload: { products: json, listId }
 });
 const createNewShoppingListSuccess = data => ({
-  type: ShoppingListActionTypes.ADD_SHOPPING_LIST_SUCCESS,
+  type: ShoppingListActionTypes.ADD_LIST_SUCCESS,
```
Comment (alemikolo) [inline client/src/modules/shopping-list/model/actions.js]: action type should be: ADD_LIST_SUCCESS or ADD_SUCCESS?

File: client/src/modules/shopping-list/model/actions.js (hunk 2 above)
Comment (alemikolo) [reply client/src/modules/shopping-list/model/actions.js]: why json[0]?

**Pull request merged**